internal/handlers/header_credentials_handler.go

```diff
@@ -0,0 +1,49 @@
+package handlers
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/ydataai/authentication-service/internal/configurations"
+	authErrors "github.com/ydataai/authentication-service/internal/errors"
+	"github.com/ydataai/go-core/pkg/common/logging"
+)
+
+// HeaderCredentialsHandler defines a HeaderCredentialsHandler struct.
+type HeaderCredentialsHandler struct {
+	restCtrlConfig configurations.RESTControllerConfiguration
```

Comment:
why is this handler receiving the controller configuration? why did we change this?

Reply:
it was because of AuthHeader, like env var... but I'll remove this.